src/app/store/reducers/courses.reducer.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    ActionReducer,
+    ActionReducerMap,
+    createFeatureSelector,
+    createSelector,
+    MetaReducer,
+    State
+} from '@ngrx/store';
+import { environment } from '../../../environments/environment';
+
+import * as coursesActions from '../actions/courses.action';
+import { CoursesState } from '../../shared/models/course/coursesState';
+
+export const coursesFeatureKey = 'courses';
+
+export const initialState: CoursesState = {
+    courses: [
+        {
+            title: 'Lorem Ipsum-11',
+            description: 'testn111',
+            creationDate: '2019-01-06T22:00:00.000Z',
+            duration: '87',
+            id: 20
+        },
+        {
+            title: 'Lorem21 412 412 412 41 Ipsum-11',
+            description: 'testn112 421 4 124 21 412 41',
+            creationDate: '2019-01-06T22:00:00.000Z',
+            duration: '2412',
+            id: 21
+        },
+    ],
+    loaded: false,
+    loading: false
+};
+
+export function reducer(state = initialState, action: any): CoursesState {
+    switch ( action.type ) {
+        case coursesActions.GetAllCourse: {
+            return {
+                ...state,
+                loading: true,
```

Comment:
Remove those "loading", "loaded" properties as we don't  use them anywhere anymore.